rueidiscompat/command.go

```diff
@@ -5107,3 +5107,189 @@ func newSlowLogCmd(res rueidis.RedisResult) *SlowLogCmd {
 	cmd.from(res)
 	return cmd
 }
+
+// LCSQuery is a parameter used for the LCS command
+type LCSQuery struct {
+	Key1         string
+	Key2         string
+	Len          bool
+	Idx          bool
+	MinMatchLen  int
+	WithMatchLen bool
+}
+
+// LCSMatch is the result set of the LCS command
+type LCSMatch struct {
+	MatchString string
+	Matches     []LCSMatchedPosition
+	Len         int64
+}
+
+type LCSMatchedPosition struct {
+	Key1 LCSPosition
+	Key2 LCSPosition
+
+	// only for withMatchLen is true
+	MatchLen int64
+}
+
+type LCSPosition struct {
+	Start int64
+	End   int64
+}
+
+type LCSCmd struct {
+	baseCmd[*LCSMatch]
+
+	// 1: match string
+	// 2: match len
+	// 3: match idx LCSMatch
+	readType uint8
+}
+
+func newLCSCmd(res rueidis.RedisResult, readType uint8) *LCSCmd {
+	cmd := &LCSCmd{readType: readType}
+	cmd.from(res)
+	return cmd
+}
+
+func (cmd *LCSCmd) SetVal(val *LCSMatch) {
+	cmd.val = val
+}
+
+func (cmd *LCSCmd) SetErr(err error) {
+	cmd.err = err
+}
+
+func (cmd *LCSCmd) Val() *LCSMatch {
+	return cmd.val
+}
+
+func (cmd *LCSCmd) Err() error {
+	return cmd.err
+}
+
+func (cmd *LCSCmd) Result() (*LCSMatch, error) {
+	return cmd.val, cmd.err
+}
+
+func (cmd *LCSCmd) from(res rueidis.RedisResult) {
+	lcs := &LCSMatch{}
+	var err error
+
+	switch cmd.readType {
+	case 1:
+		// match string
+		if lcs.MatchString, err = res.ToString(); err == nil {
```

Comment:
```suggestion
		if lcs.MatchString, err = res.ToString(); err != nil {
```
Shouldn't this be `err != nil`?